.github/workflows/pkgdown.yaml

```diff
@@ -38,38 +38,24 @@ jobs:
           extra-repositories: "https://mc-stan.org/r-packages/"
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
-          working-directory: wwinference
           cache: true
           pak-version: rc
           extra-packages: any::pkgdown, local::.
           needs: website
+          
       - name: "Install cmdstan via cmdstanr"
         uses: epinowcast/actions/install-cmdstan@v1
         with:
           cmdstan-version: "latest"
 
       - name: Build site
-        run: "pkgdown::build_site_github_pages('wwinference', new_process = FALSE, install = FALSE)"
+        run: "pkgdown::build_site_github_pages(new_process = FALSE, install = FALSE)"
         shell: Rscript {0}
 
-      - name: Upload artifact for GH pages deployment
-        uses: actions/upload-pages-artifact@v2
+      - name: Deploy to GitHub pages 🚀
+        if: github.event_name != 'pull_request'
+        uses: JamesIves/github-pages-deploy-action@v4.5.0
         with:
-          path: "wwinference/docs/"
-
-    deploy:
-      # check builds on PRs but only deploy when main changes
-      if: ${{ github.event_name != 'pull_request' }}
-      needs: build
-      runs-on: ubuntu-latest
-      permissions:
-        pages: write
-        id-token: write
-
-      environment:
-        name: github-pages
-        url: ${{ steps.deployment.outputs.page_url }}
-
-      steps:
-        - name: Deploy to GitHub pages
-          uses: actions/deploy-pages@v2.0.4
+          clean: false
+          branch: gh-pages
```

Comment:
I would use environment deployment, not deployment to a github pages branch. [Relevant Github documentation on the distinction](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site).

Here's an example of an action for environment deployment: https://github.com/CDCgov/wastewater-informed-covid-forecasting/blob/prod/.github/workflows/pkgdown.yaml

Reply:
Ok repeating back in other words, we want to trigger a change to the page whenever a PR is merged into main, we don't want to make a new branch called `gh-pages`  and say have to merge to that to get updates

Reply:
First sentence is correct. RE: second: an older Github Pages workflow was to store rendered website files in a dedicated branch  in the repo (typically named `gh-pages`). That has now been superseded by an approach where they're stored outside the repo itself. One advantage of this second approach is that it more clearly separates the pre-rendered source (stored in the repo, version-controlled) from the rendered website files (stored outside it, not necessarily version-controlled)